smac_planner/src/costmap_downsampler.cpp

```diff
@@ -74,10 +90,7 @@ nav2_costmap_2d::Costmap2D * CostmapDownsampler::downsample(
     }
   }
 
-  if (_node->count_subscribers(_topic_name) > 0) {
-    _downsampled_costmap_pub->publishCostmap();
-  }
-
+  _downsampled_costmap_pub->publishCostmap();
```

Comment:
This should still only publish when someone is listening. Maybe this is something that the costmap publisher object can do? It itself has a publisher under the hood and the publisher can check for the number of subscribers rather than the node

Reply:
Exactly, `Costmap2DPublisher::publishCostmap()` already checks for subscription count and publishes only when someone is listening.

Reply:
Oh nice, ok